src/Sets/Zonotope.jl

```diff
@@ -329,30 +329,149 @@ function scale!(α::Real, Z::Zonotope)
     return Z
 end
 
-"""
-    reduce_order(Z::Zonotope, r::Union{Integer, Rational})
+# ==================================
+# Zonotope order reduction methods
+# ==================================
+
+# These methods split the given zonotope Z into two zonotopes, K and L, where
+# K contains the most "representative" generators and L contains the generators
+# that are reduced, Lred using a box overapproximation
+struct GIR05 <: AbstractReductionMethod end
+struct COMB03 <: AbstractReductionMethod end
+
+const _GIR05 = GIR05()
+const _COMB03 = COMB03()
+
+# Implements zonotope order reduction method from [COMB03]
+# We follow the notation from [YS18]
+function _reduce_order(Z::Zonotope, r::Number, ::COMB03)
+    r >= 1 || throw(ArgumentError("the target order should be at least 1, but it is $r"))
+    c = Z.center
+    G = Z.generators
+    n, p = size(G)
 
-Reduce the order of a zonotope by overapproximating with a zonotope with fewer
-generators.
+    # r is bigger than the order of Z => don't reduce
+    (r * n >= p) && return Z
 
-### Input
+    # this algorithm sort generators by decreasing 2-norm
+    indices = Vector{Int}(undef, p)
+    _weighted_gens!(indices, G, _COMB03)
 
-- `Z` -- zonotope
-- `r` -- desired order
+    # the first m generators have greatest weight
+    m = floor(Int, n * (r - 1))
 
-### Output
+    # compute interval hull of L
+    Lred = _interval_hull(G, view(indices, (m+1):p))
 
-A new zonotope with fewer generators, if possible.
+    isone(r) && return Zonotope(c, Lred)
 
-### Algorithm
+    Gred = _hcat_KLred(G, view(indices, 1:m), Lred)
+    return Zonotope(c, Gred)
+end
 
-See `overapproximate(Z::Zonotope, ::Type{<:Zonotope}, r::Union{Integer, Rational})`
-for details.
-"""
-function reduce_order(Z::Zonotope, r::Union{Integer, Rational})
-    return overapproximate(Z, Zonotope, r)
+# Implements zonotope order reduction method from [GIR05]
+# We follow the notation from [YS18]
+function _reduce_order(Z::Zonotope, r::Number, ::GIR05)
+    r >= 1 || throw(ArgumentError("the target order should be at least 1, but it is $r"))
+    c = Z.center
+    G = Z.generators
+    n, p = size(G)
+
+    # r is bigger than the order of Z => don't reduce
+    (r * n >= p) && return Z
+
+    # this algorithm sorts generators by ||⋅||₁ - ||⋅||∞ difference
+    indices = Vector{Int}(undef, p)
+    _weighted_gens!(indices, G, _GIR05)
+
+    # the first m generators have greatest weight
+    m = floor(Int, n * (r - 1))
+
+    # compute interval hull of L
+    Lred = _interval_hull(G, view(indices, (m+1):p))
+
+    isone(r) && return Zonotope(c, Lred)
+
+    Gred = _hcat_KLred(G, view(indices, 1:m), Lred)
+    return Zonotope(c, Gred)
+end
```

Comment:
The two methods `_reduce_order` are identical if you pass the algorithm to `_weighted_gens!`. So they should be merged.